crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -573,6 +654,38 @@ impl BuildSpec {
 		chain_spec.to_file(&self.output_file)?;
 		Ok(())
 	}
+
+	// Generates the deterministic runtime and returns the runtime generated.
+	fn generate_deterministic_runtime(
+		&self,
+		cli: &mut impl cli::traits::Cli,
+	) -> anyhow::Result<Vec<u8>> {
+		let engine = ContainerEngine::detect()?;
+		// As srtool-cli(https://github.com/chevdor/srtool-cli/blob/master/cli/src/main.rs#L28) warn about using docker.
+		if engine == ContainerEngine::Docker {
+			cli.warning("WARNING: You are using docker. We recommend using podman instead.")?;
+		}
+		let builder = SrToolBuilder::new(
+			engine,
+			None,
+			self.package.clone(),
+			self.profile.clone(),
+			self.runtime_dir.clone(),
+		)?;
+		let wasm_path = builder.generate_deterministic_runtime()?;
```

Comment:
I see now why you used 'generate'

Reply:
https://github.com/r0gue-io/pop-cli/pull/425/commits/c7dea912f7efe5ac15c68d47e13e6dfc6c2b85cc